superglue/lib/utils/ujs.ts

```diff
@@ -17,21 +34,14 @@ export class HandlerBuilder {
     this.visitOrRemote = this.visitOrRemote.bind(this)
   }
 
-  retrieveLink(target) {
-    let link = target
-    while (!!link.parentNode && link.nodeName.toLowerCase() !== 'a') {
-      link = link.parentNode
-    }
-
-    if (
-      link.nodeName.toLowerCase() === 'a' &&
-      link.href.length !== 0
-    ) {
+  retrieveLink(target: Element): HTMLAnchorElement | undefined {
+    const link = target.closest<HTMLAnchorElement>('a')
```

Comment:
This may have been a turbolinks 3 thing. `closest` is what I should be using now